vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -32,14 +32,14 @@ def save_inventory_prep(find_key, child_keys, extra_keys)
     return find_key, child_keys, extra_keys, remove_keys
   end
 
-  def save_inventory(type, klass, parent, hash, deletes, new_records, record_index, record_index_columns, find_key, child_keys, remove_keys)
+  def save_inventory(type, parent, hash, deletes, new_records, record_index, record_index_columns, find_key, child_keys, remove_keys)
     # Backup keys that cannot be written directly to the database
     key_backup = backup_keys(hash, remove_keys)
 
     # Find the record, and update if found, else create it
     found = find_key.blank? ? parent.send(type) : self.save_inventory_record_index_fetch(record_index, record_index_columns, hash, find_key)
     if found.nil?
-      found = klass.new(hash)
+      found = find_key ? parent.send(type).build(hash) : parent.send("build_#{type}", hash)
```

Comment:
I was thinking this find_key call would come out and be new_records.nil?  because using find_key as the multi identifier is really the "change 3" from before.

Do all multis have find_keys?